src/containers/Tenant/Schema/CreateDirectoryDialog/CreateDirectoryDialog.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+
+import {Dialog, TextInput} from '@gravity-ui/uikit';
+
+import {ResponseError} from '../../../../components/Errors/ResponseError';
+import {schemaApi} from '../../../../store/reducers/schema/schema';
+import {cn} from '../../../../utils/cn';
+import i18n from '../../i18n';
+
+import './CreateDirectoryDialog.scss';
+
+const b = cn('ydb-schema-create-directory-dialog');
+
+interface SchemaTreeProps {
+    open: boolean;
+    onClose: VoidFunction;
+    parentPath: string;
+    onSuccess: (value: string) => void;
+}
+
+function validateRelativePath(value: string) {
+    if (value && /\s/.test(value)) {
+        return i18n('schema.tree.dialog.whitespace');
+    }
+    return '';
+}
+
+export function CreateDirectoryDialog({open, onClose, parentPath, onSuccess}: SchemaTreeProps) {
+    const [error, setError] = React.useState<unknown>('');
+    const [relativePath, setRelativePath] = React.useState('');
+    const [create, response] = schemaApi.useCreateDirectoryMutation();
+
+    const handleUpdate = (updated: string) => {
+        setRelativePath(updated);
+        setError(validateRelativePath(updated));
```

Comment:
Do we really need to validate field on change?

Reply:
Actually no